tests/test_validation.py

```diff
@@ -120,7 +120,22 @@ def test_user_creation_validates():
         ({'email_address': 'this@that.com',
           'role': 'invalid',
           'name': exactly_255,
-          'password': exactly_255}, False, "invalid role")
+          'password': exactly_255}, False, "invalid role"),
+        ({'email_address': 'this@that.com',
+          'role': 'buyer',
+          'name': exactly_255,
+          'supplier_id': 123,
+          'password': exactly_255}, True, "valid supplier id"),
+        ({'email_address': 'this@that.com',
+          'role': 'buyer',
+          'name': exactly_255,
+          'supplier_id': '',
+          'password': exactly_255}, False, "invalid supplier id (to short)"),
```

Comment:
These last two don't really make sense - supplier ID is numeric, so whether they are supplied as a long or short string should make no difference: if it's a string of any length then it's invalid.